### PR TITLE
Update for Sleuth api review

### PR DESCRIPTION
**Pull request by moarychan**
Add a gist for Spring Cloud Azure Sleuth usage:
https://gist.github.com/moarychan/78cd8312085335230722493c3d037714

Fix code smells during the review.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java [hunk 1]
```diff
@@ -38,12 +38,13 @@ public void setBeanFactory(BeanFactory beanFactory) {
...
-            HttpPipelinePolicy policy = beanFactory.getBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME, HttpPipelinePolicy.class);
-            if (policy != null) {
+            try {
+                HttpPipelinePolicy policy = beanFactory.getBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME,
+                    HttpPipelinePolicy.class);
                 AbstractAzureHttpClientBuilderFactory builderFactory = (AbstractAzureHttpClientBuilderFactory) bean;
                 builderFactory.addHttpPipelinePolicy(policy);
                 LOGGER.debug("Added the Sleuth http pipeline policy to {} builder.", bean.getClass());
-            } else {
+            } catch (BeansException exception) {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java:47]: Why don't we use beanFactory.containsBean here?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java (hunk 1 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java:47]: Use bean name and bean class name safer, is it necessary?

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java]: I don't get your question here.

File: sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java [hunk 2]
```diff
@@ -37,16 +37,12 @@ public void setBeanFactory(BeanFactory beanFactory) {
...
-            try {
-                HttpPipelinePolicy policy = beanFactory.getBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME,
-                    HttpPipelinePolicy.class);
-                AbstractAzureHttpClientBuilderFactory builderFactory = (AbstractAzureHttpClientBuilderFactory) bean;
-                builderFactory.addHttpPipelinePolicy(policy);
-                LOGGER.debug("Added the Sleuth http pipeline policy to {} builder.", bean.getClass());
-            } catch (BeansException exception) {
-                LOGGER.warn("Not found the Sleuth http pipeline policy for {} builder.", bean.getClass());
-            }
+        if (bean instanceof AbstractAzureHttpClientBuilderFactory
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java:40]: Can we change here to 
```java
if (!bean instanceof xxx) {
return bean;
}
```

Comment (moarychan) [reply sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java]: OK

File: sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java [hunk 3]
```diff
@@ -35,14 +35,19 @@ public void setBeanFactory(BeanFactory beanFactory) {
...
+            return bean;
+        }
+
+        if (beanFactory.containsBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME)) {
             HttpPipelinePolicy policy = (HttpPipelinePolicy) beanFactory.getBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME);
             AbstractAzureHttpClientBuilderFactory builderFactory = (AbstractAzureHttpClientBuilderFactory) bean;
             builderFactory.addHttpPipelinePolicy(policy);
             LOGGER.debug("Added the Sleuth http pipeline policy to {} builder.", bean.getClass());
+        } else {
+            LOGGER.warn("Not found the Sleuth http pipeline policy for {} builder.", bean.getClass());
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java:50]: Actually, have you tried that if there's no sleuth pipeline bean in the beanFactory, how many times will this warning log be printed?

**Review by saragluna — APPROVED** [review]
LGTM.

**Comment by saragluna** [comment]
Merging this pr for the ci failed for pipeline issues.

**Pull request merged**